tests/metrics/genai/test_genai_metrics.py

```diff
@@ -1219,3 +1209,111 @@ def test_log_make_genai_metric_fn_args():
+
+
+def test_genai_metrics_callable(custom_metric):
+    data = {
+        "predictions": mlflow_prediction,
+        "inputs": "What is MLflow?",
+        "targets": mlflow_ground_truth,
+    }
+    with mock.patch.object(
+        model_utils,
+        "score_model_on_payload",
+        return_value=properly_formatted_openai_response1,
+    ):
+        expected_result = custom_metric.eval_fn(
+            pd.Series([mlflow_prediction]),
+            {},
+            pd.Series(["What is MLflow?"]),
+            pd.Series([mlflow_ground_truth]),
+        )
+        metric_value = custom_metric(**data)
+        scores = custom_metric(
+            **data,
+            return_only_scores=True,
+        )
+
+    assert metric_value == expected_result
+    assert metric_value.scores == [3]
+    assert metric_value.justifications == [openai_justification1]
+    assert metric_value.aggregate_results == {
+        "mean": 3,
+        "variance": 0,
+        "p90": 3,
+    }
+    assert scores == [3]
+    assert set(inspect.signature(custom_metric).parameters.keys()) == {
+        "predictions",
+        "inputs",
+        "metrics",
+        "return_only_scores",
+        "targets",
+    }
+
+
+def test_genai_metrics_callable_errors(custom_metric):
+    with pytest.raises(TypeError, match=r"missing 1 required keyword-only argument: 'inputs'"):
+        custom_metric(predictions=mlflow_prediction)
+
+    data = {
+        "predictions": mlflow_prediction,
+        "inputs": "What is MLflow?",
+    }
+    with pytest.raises(MlflowException, match=r"Missing required arguments: {'targets'}"):
+        custom_metric(**data)
+
+    with pytest.raises(MlflowException, match=r"Unexpected arguments: {'data'}"):
+        custom_metric(**data, targets=mlflow_ground_truth, data="data")
+
+    with pytest.raises(TypeError, match=r"Expected predictions to be a string or list"):
+        custom_metric(predictions=1, inputs="What is MLflow?", targets=mlflow_ground_truth)
+
+
+def test_genai_metrics_with_llm_judge_callable():
+    custom_judge_prompt = "This is a custom judge prompt that uses {input} and {output}"
+
+    custom_judge_prompt_metric = make_genai_metric_from_prompt(
+        name="custom",
+        judge_prompt=custom_judge_prompt,
+        metric_metadata={"metadata_field": "metadata_value"},
+    )
+
+    inputs = ["What is MLflow?", "What is Spark?"]
+    outputs = [
+        "MLflow is an open-source platform",
+        "Apache Spark is an open-source distributed framework",
+    ]
+
+    with mock.patch.object(
+        model_utils,
+        "score_model_on_payload",
+        return_value=properly_formatted_openai_response1,
+    ):
+        expected_result = custom_judge_prompt_metric.eval_fn(
+            input=pd.Series(inputs), output=pd.Series(outputs)
+        )
+        metric_value = custom_judge_prompt_metric(
+            input=pd.Series(inputs),
+            output=pd.Series(outputs),
+        )
```

Comment:
Can we also support scalars so that users don't have to wrap these arguments in `pd.Series` if there's only one item?